mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1319,10 +1331,28 @@ void CppEmitter::cacheDeferredOpResult(Value value, StringRef str) {
 /// Return the existing or a new name for a Value.
 StringRef CppEmitter::getOrCreateName(Value val) {
```

Comment:
`getOrCreateName` seems to be specifically about assigning variable names to `mlir::Values`. As such, it seems odd to have the code to emit constants values here. 
With this change, the distinction between `emitOperand` and `getOrCreateName` is very blurry.

I think I would prefer to call `emitOperand` from the emitter of the SubscriptOp. `getOrCreateName` should only be used if we need an lvalue / variable, e.g. in the right-hand-side of assignments.

Reply:
Note that the code is now different but my new workaround still makes use of the lvalue/deferred ops cache to store the ConstantOp string.